core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/FixedRateSamplingTelemetryProcessor.java

```diff
@@ -0,0 +1,175 @@
+
+import com.microsoft.applicationinsights.agent.internal.common.StringUtils;
+import com.microsoft.applicationinsights.extensibility.TelemetryProcessor;
+import com.microsoft.applicationinsights.internal.annotation.BuiltInProcessor;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.telemetry.SupportSampling;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Created by Dhaval Doshi Oct 2017
+ * This processor is used to Perform Sampling on User specified sampling rate
+ */
+@BuiltInProcessor("FixedRateSamplingTelemetryProcessor")
+public final class FixedRateSamplingTelemetryProcessor implements TelemetryProcessor {
+
+    private final String dependencyTelemetryName = "Dependency";
+    private static final String eventTelemetryName = "Event";
+    private static final String exceptionTelemetryName = "Exception";
+    private static final String pageViewTelemetryName = "PageView";
+    private static final String requestTelemetryName = "Request";
+    private static final String traceTelemetryName = "Trace";
+
+    private static final String listSeparator = ";";
+    private static Map<String, Class> allowedTypes;
+
+    private String excludedTypesString;
+    private Set<Class> excludedTypesHashSet;
+
+    private String includedTypesString;
+    private Set<Class> includedTypesHashSet;
+
+    public double samplingPercentage;
+
+    /**
+     * constructor is responsible of initializing this processor
+     * to default settings
+     */
+    public FixedRateSamplingTelemetryProcessor() {
+        this.samplingPercentage = 100.00;
+        this.samplingPercentage = samplingPercentage;
+        this.includedTypesHashSet = new HashSet<Class>();
+        this.excludedTypesHashSet = new HashSet<Class>();
+        try {
+            this.allowedTypes = new HashMap<String, Class>() {{
+                put(dependencyTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.RemoteDependencyTelemetry"));
+                put(eventTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.EventTelemetry"));
+                put(exceptionTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.ExceptionTelemetry"));
+                put(pageViewTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.PageViewTelemetry"));
+                put(requestTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.RequestTelemetry"));
+                put(traceTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.TraceTelemetry"));
+            }};
+        }
+        catch (ClassNotFoundException e) {
+            InternalLogger.INSTANCE.trace("Unable to locate telemetry classes");
+        }
+    }
+
+    public String getExcludedTypesString() {
```

Comment:
why anybody would need to get a string, not the parsed SET?

Reply:
Ideally no one should need it. You are right this getters can be removed.